cmd/main.go

```diff
@@ -72,11 +72,15 @@ func startServer(config *config.Config, airbrakeNotifier *gobrake.Notifier, logg
 		return err
 	}
 
-	if server.TLSConfig, err = config.ExtractServiceTLSConfig(logger); err != nil {
-		return err
+	if config.DisableTLS {
+		err = server.ListenAndServe()
```

Comment:
fleet-telemetry relies upon the device's certificate for its identify. With this implementation, won't `extractCertFromHeaders` panic since `r.TLS` would be `nil`?

https://github.com/teslamotors/fleet-telemetry/blob/c6050edf0cea815487b462f75ec72f026d0851d2/server/streaming/server.go#L237-L244